OMEdit/OMEditLIB/FMI/FMUExportOutputWidget.cpp

```diff
@@ -58,17 +59,25 @@ FmuExportOutputWidget::FmuExportOutputWidget(LibraryTreeItem* pLibraryTreeItem,
      * fix issue https://github.com/OpenModelica/OpenModelica/issues/12916,
      * use hashed string for fmu tmp directory
     */
-    QString hashedString = QString::number(stringHashDjb2(mmc_mk_scon(OptionsDialog::instance()->getFMIPage()->getFMUNameTextBox()->text().toUtf8().constData())));
-    mFmuTmpPath = QDir::currentPath().append("/").append(hashedString.left(3)+".fmutmp");
+    if (mTargetLanguage.compare("C") == 0) {
+      QString hashedString = QString::number(stringHashDjb2(mmc_mk_scon(OptionsDialog::instance()->getFMIPage()->getFMUNameTextBox()->text().toUtf8().constData())));
+      mFmuTmpPath = QDir::currentPath().append("/").append(hashedString.left(3)+".fmutmp");
+    } else {
+      mFmuTmpPath = QDir::currentPath();
+    }
     mFMUName = OptionsDialog::instance()->getFMIPage()->getFMUNameTextBox()->text();
   } else {
     /*
      * fix issue https://github.com/OpenModelica/OpenModelica/issues/12916,
      * use hashed string for fmu tmp directory
      */
-    QString hashedString = QString::number(stringHashDjb2(mmc_mk_scon(mpLibraryTreeItem->getName().toUtf8().constData())));
-    mFmuTmpPath = QDir::currentPath().append("/").append(hashedString.left(3)+".fmutmp");
-    mFMUName = mpLibraryTreeItem->getName();
+     if (mTargetLanguage.compare("C") == 0) {
+       QString hashedString = QString::number(stringHashDjb2(mmc_mk_scon(mpLibraryTreeItem->getName().toUtf8().constData())));
+       mFmuTmpPath = QDir::currentPath().append("/").append(hashedString.left(3)+".fmutmp");
+     } else {
+       mFmuTmpPath = QDir::currentPath();
+     }
+     mFMUName = mpLibraryTreeItem->getName();
```

Comment:
Can we improve this code? Instead of repeating the same code we can do it once.

Do this,
```
const QString fmuNameText = OptionsDialog::instance()->getFMIPage()->getFMUNameTextBox()->text();
mFMUName = fmuNameText.isEmpty() ? mpLibraryTreeItem->getName() : fmuNameText;
```

And then use `mFMUName` to generate the hash string.

Reply:
fixed with https://github.com/OpenModelica/OpenModelica/pull/13142/commits/79395e5b718fcd2262996f8df9dc5b1fc1783abb